### PR TITLE
Fix the documentation of the CNOTDihedral operator class

### PR DESCRIPTION
**Pull request by ShellyGarion**
### Summary

Minor fix of the documentation of the `CNOTDihedral` class,
and `tensor` and `expand` methods (also for the `Clifford` class).

### CODE REVIEW

File: qiskit/quantum_info/operators/dihedral/dihedral.py [hunk 1]
```diff
@@ -33,19 +33,19 @@ class CNOTDihedral(BaseOperator, AdjointMixin):
...
+    and :class:`~qiskit.circuit.library.XGate`.
 
     **Representation**
 
     An *N*-qubit CNOT-Dihedral operator is stored as an affine function and a
     phase polynomial, based on the convention in references [1, 2].
 
-    The affine function consists of an NxN invertible binary matrix,
-    and an N binary vector.
+    The affine function consists of an *NxN* invertible binary matrix,
```
Comment (mtreinish) [inline qiskit/quantum_info/operators/dihedral/dihedral.py:44]: A single `*` is italics is that your intent here? I might use mathmode/latex here, like:
```suggestion
    The affine function consists of an :math:`N \times N` invertible binary matrix,
```
Of course then you'd probably want to change the `*N*` elsewhere to be:
```
:math:`N`
```

**Review by mtreinish — COMMENTED** [review]
LGTM, just one question inline

File: qiskit/quantum_info/operators/dihedral/dihedral.py (hunk 1 above)
Comment (ShellyGarion) [reply qiskit/quantum_info/operators/dihedral/dihedral.py]: Thanks @mtreinish - I updated the docs accordingly

**Comment by mtreinish** [comment]
Thanks for updating, I think I left a bad inline suggestion though, we need to use `\\` or put `r"""` on the front of the docstring. Since it's not a raw string it's escaping the `\t` and making it a tab, so things don't render correctly:

Screenshot_2021-04-12_08-03-06

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for the quick update.

**Pull request merged**